src/internal/operators/takeWhile.ts

```diff
@@ -57,12 +57,34 @@ export function takeWhile<T>(predicate: (value: T, index: number) => boolean, in
 export function takeWhile<T>(predicate: (value: T, index: number) => boolean, inclusive = false): MonoTypeOperatorFunction<T> {
   return operate((source, subscriber) => {
     let index = 0;
+    // Whether or not we're still taking values.
+    let taking = true;
     source.subscribe(
-      new OperatorSubscriber(subscriber, (value) => {
-        const result = predicate(value, index++);
-        (result || inclusive) && subscriber.next(value);
-        !result && subscriber.complete();
-      })
+      new OperatorSubscriber(
+        subscriber,
+        (value) => {
+          taking = predicate(value, index++);
+          if (taking || inclusive) {
+            // If we're still taking, Or if this was "inclusive", meaning
+            // we found the "last" value, then we send the value along.
+            subscriber.next(value);
+          }
+          if (!taking) {
+            // We're no longer taking values after calling the predicate above
+            // time to complete the result.
+            subscriber.complete();
+          }
+        },
+        undefined,
```

Comment:
we should probably also block complete notifications here, as was done in take

Reply:
This has been added.